src/agent/src/message_queue_utils.cpp

```diff
@@ -48,3 +50,44 @@ void PushCommandsToQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue, const
         }
     }
 }
+
+std::optional<command_store::Command> GetCommandFromQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue)
+{
+    if (multiTypeQueue->isEmpty(MessageType::COMMAND))
+    {
+        return std::nullopt;
+    }
+
+    Message m = multiTypeQueue->getNext(MessageType::COMMAND);
+    nlohmann::json jasonData = m.data.at(0).at("data");
+
+    std::string strId = jasonData["id"].dump();
+
+    // remove quotation marks from id
+    strId.erase(strId.begin());
+    strId.erase(strId.end() - 1);
+    int64_t id = std::stoll(strId);
+
+    std::string module = jasonData["origin"]["module"].dump();
+    std::string command = jasonData.contains("command") ? jasonData["command"].dump() : "";
+    std::string parameters = jasonData.contains("parameters") ? jasonData["parameters"].dump() : "";
+    std::string result = "Pending execution";
+    command_store::Status status = command_store::Status::IN_PROGRESS;
+
+    command_store::Command cmd(id, module, command, parameters, result, status);
+
+    return cmd;
+}
+
+bool PopCommandFromQueue(std::shared_ptr<IMultiTypeQueue> multiTypeQueue)
+{
+    return multiTypeQueue->pop(MessageType::COMMAND);
+}
+
+int DispatchCommand(command_store::Command& cmd)
+{
+    // TO_DO: Implement real dispatch function.
+    std::cout << "Dispatching command " << cmd.m_module << "\n";
+    std::this_thread::sleep_for(std::chrono::milliseconds(DISPATCH_WAIT_MSECS));
```

Comment:
Remove this line, won't be necessary. Also remove https://github.com/wazuh/wazuh-agent/pull/150/files#diff-6e0f7ef87f491f91b5469ed4abda6d6d12f1470d38e6a0b26c6a7997fd4aa773R7 and https://github.com/wazuh/wazuh-agent/pull/150/files#diff-6e0f7ef87f491f91b5469ed4abda6d6d12f1470d38e6a0b26c6a7997fd4aa773R14.

Reply:
Done